### PR TITLE
File property

### PR DESCRIPTION
**Pull request by fwkoch**

### CODE REVIEW

**Comment by bsmithyman** [comment]
I think that PR :100: looks okay to me. You thinking to get this into `dev` after the release of the current `dev` to `master`?

File: properties/basic.py [hunk 1]
```diff
@@ -760,6 +760,109 @@ def from_json(value, **kwargs):
...
+                            'valid_modes'.format(self.mode))
+        for val in value:
+            if val not in FILE_MODES:
+                raise TypeError('Invalid file mode: {}'.format(val))
+        self._valid_mode = tuple(value)
+
+    def get_property(self):
+        """Establishes access of Property values"""
+
+        prp = super(File, self).get_property()
```
Comment (ThomasMatern) [inline properties/basic.py]: Can you please rename this to `prop`

File: properties/basic.py [hunk 2]
```diff
@@ -760,6 +760,109 @@ def from_json(value, **kwargs):
...
+
+        If value is a string, it attempts to open it with the given mode.
+        """
+        if isinstance(value, string_types) and self.mode is not None:
+            try:
+                value = open(value, self.mode)
+            except (IOError, TypeError):
+                self.error(instance, value,
+                           extra='Cannot open file: {}'.format(value))
+        if not all([hasattr(value, att) for att in ('read', 'seek')]):
```
Comment (ThomasMatern) [inline properties/basic.py]: ... and this one to `attr`

**Comment by fwkoch** [comment]
Yes, this will go into `dev` as the first new feature for v0.3.1. Thanks for reviewing!

File: properties/basic.py (hunk 1 above)
Comment (fwkoch) [inline properties/basic.py]: @ThomasMatern - Sorry for the wrong tag in my initial (unedited) comment.

**Comment by bsmithyman** [comment]
@fwkoch: Thoughts on merging this in?

**Pull request merged**